### PR TITLE
Feature/pt 34/setup vite react

### LINKED ISSUE #34: Setup react app
- To consume the api
- Use vite

### PR DESCRIPTION
**Pull request by samresh9**
Fixes #34 
- setup react vite app with default home page
- setup shared eslint config between apps

### CODE REVIEW

File: packages/eslint-config-custom/index.js [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+    "plugin:react-hooks/recommended",
+  ],
   rules: {
-    "@next/next/no-html-link-for-pages": "off",
+    "prettier/prettier": [
+      "error",
+      {
+        endOfLine: "auto",
+      },
+    ],
```
Comment (therj) [inline packages/eslint-config-custom/index.js:16]: Move all rules, not just prettier.

File: packages/eslint-config-custom/index.js [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+    "import/no-extraneous-dependencies": ["error", { devDependencies: true }],
   },
```
Comment (therj) [inline packages/eslint-config-custom/index.js]: Why should devDependencies be allowed?

**Review by therj — CHANGES_REQUESTED** [review]

File: packages/eslint-config-custom/index.js (hunk 1 above)
Comment (samresh9) [reply packages/eslint-config-custom/index.js:16]: Its not working, when I moved all the rules to the shared config the rules are not extended and error occurs.

File: packages/eslint-config-custom/index.js [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
-  extends: ["next", "turbo", "prettier"],
+  env: {
+    browser: true,
+    commonjs: true,
+    node: true,
+    jest: true,
+  },
+  ignorePatterns: ["**/dist/*"],
+  extends: ["airbnb-base", "plugin:prettier/recommended", "prettier", "turbo"],
+  plugins: ["react-refresh", "import"],
```
Comment (therj) [inline packages/eslint-config-custom/index.js]: These plugins got moved to frontend.

File: packages/eslint-config-custom/index.js [hunk 4]
```diff
@@ -1,11 +1,38 @@
...
+    "no-underscore-dangle": ["error", { allow: ["_id"] }],
+    quotes: [
+      "error",
+      "double",
+      { avoidEscape: true, allowTemplateLiterals: true },
+    ],
+    "react-refresh/only-export-components": [
+      "warn",
+      { allowConstantExport: true },
+    ],
```
Comment (therj) [inline packages/eslint-config-custom/index.js]: This should be front end specific

File: apps/frontend/.eslintrc.cjs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  root: true,
+  env: { browser: true, es2020: true },
+  extends: [
+    "plugin:react/recommended",
+    "plugin:react/jsx-runtime",
+    "plugin:react-hooks/recommended",
+    "custom",
+  ],
+  ignorePatterns: ["dist"],
```
Comment (therj) [inline apps/frontend/.eslintrc.cjs]: This is already in customconfig

**Review by therj — CHANGES_REQUESTED** [review]
changes required around es lint misconfig.

File: apps/frontend/.eslintrc.cjs [hunk 6]
```diff
@@ -7,15 +7,15 @@ module.exports = {
...
   ],
-  ignorePatterns: ["dist"],
+  plugins: ["react-refresh"],
   parserOptions: {
     ecmaVersion: 2022,
     sourceType: "module",
   },
-  settings: {
-    react: { version: "18.2" },
-  },
```
Comment (therj) [inline apps/frontend/.eslintrc.cjs]: Where did the react version go?

Comment (therj) [reply apps/frontend/.eslintrc.cjs]: "eslint-plugin-import": "^2.27.5", should come to frontend plugins (is it required?)

Comment (therj) [reply apps/frontend/.eslintrc.cjs]: react settings went to shared config, why?

Comment (samresh9) [reply apps/frontend/.eslintrc.cjs]: > -
previously it was showing warning in lint but now reinstalling all the packages fixed it
so moved back to frontend.

File: packages/eslint-config-custom/package.json [hunk 7]
```diff
@@ -4,9 +4,10 @@
   "main": "index.js",
   "license": "MIT",
   "dependencies": {
+    "eslint-config-airbnb-base": "^15.0.0",
     "eslint-config-prettier": "^8.3.0",
     "eslint-config-turbo": "^1.9.3",
-    "eslint-config-airbnb-base": "^15.0.0",
+    "eslint-plugin-import": "^2.28.1",
```
Comment (therj) [inline packages/eslint-config-custom/package.json]: eslint-plugin-import sets up errors for ESM syntax, used at plugins: ["import"].

For now, this only affects the frontend as backend and packages use CJS.

Comment (therj) [reply packages/eslint-config-custom/package.json]: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/prefer-default-export.md

**Review by therj — APPROVED** [review]

**Pull request merged**